services/gateway/endpoints/src/main/java/org/eclipse/ditto/services/gateway/endpoints/utils/DittoRejectionHandlerFactory.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2017 Bosch Software Innovations GmbH.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/org/documents/epl-2.0/index.php
+ *
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial contribution
+ */
+package org.eclipse.ditto.services.gateway.endpoints.utils;
+
+import static akka.http.javadsl.server.Directives.complete;
+
+import java.text.MessageFormat;
+
+import akka.http.javadsl.model.StatusCodes;
+import akka.http.javadsl.server.MissingQueryParamRejection;
+import akka.http.javadsl.server.RejectionHandler;
+import akka.http.javadsl.server.Route;
+
+/**
+ * Factory to create a custom {@link akka.http.javadsl.server.RejectionHandler}.
+ */
+public final class DittoRejectionHandlerFactory {
+
+    private static final String MISSING_QUERY_PARAM_TEMPLATE = "Request is missing required query parameter '{0}'";
```

Comment:
I think that this won't work in a MessageFormat. AFAIR when you want a `'` in a string formatted by MessageFormat, you have to write it like this: `''{0}''`

Reply:
Thanks for the finding. I fixed this with commit 293b7a107e76c7684b4d19fd662ff35400fe8822.